app/models/imo_authorization.rb

```diff
@@ -0,0 +1,12 @@
+class ImoAuthorization < ApplicationRecord
+  belongs_to :patient
+  validates :last_invited_at, presence: true
+  validates :status, presence: true
+
+  enum status: {
+    invited: "invited",
+    no_imo_account: "no_imo_account",
+    subscribed: "subscribed",
```

Comment:
How do we know when they are subscribed?

Reply:
When we attempt to send an imo notification, we get a 200 if the user exists, but if the user hasn't yet subscribed, the body will be:
```    
"response": {
      "status": "failed",
      "message": "User has not subscribed",
      "error_code": "not_subscribed"
}
```
I'm not exactly sure what that body will be on a successful response because I can't test that yet and it's not in the API docs, but I'm guessing it will be different. So the general idea is that if a user's `ImoAuthorization#status` is `invited`, we will try to send them a notification via imo before sms. If the response tells us the message succeeded and doesn't tell us they're unsubscribed, we update the status to `subscribed`. Which prevents us from inviting them again.